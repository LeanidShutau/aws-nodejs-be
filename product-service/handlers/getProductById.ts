import { APIGatewayProxyHandler } from 'aws-lambda';
import 'source-map-support/register';

import { findById } from './products-db';

export const getProductById: APIGatewayProxyHandler = async (event, _context) => {
  const headers = {
    'Content-Type': 'application/json',
    'Access-Control-Allow-Origin': '*',
    'Access-Control-Allow-Headers': 'Origin, X-Requested-With, Content-Type, Accept',
    'Access-Control-Allow-Methods': 'POST, GET, OPTIONS, PATCH, PUT',
  };
  try {
    const { productId } = event.pathParameters || {};
    const product = await findById(productId);
    if (!product) {
      return {
        statusCode: 404,
        body: JSON.stringify({
          message: "Not Found",
        }, null, 2),
        headers,
      };
    }
    return {
      statusCode: 200,
      body: JSON.stringify({
        product,
      }, null, 2),
      headers,
    };
  } catch (err) {
    return {
      statusCode: 500,
      body: JSON.stringify(err, null, 2),
      headers,
    };
  }
}
