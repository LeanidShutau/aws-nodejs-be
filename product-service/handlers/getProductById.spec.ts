import { APIGatewayProxyEvent } from 'aws-lambda';
import { get as getProductById } from './getProductById';

it('should response with a product', async () => {
    const result = await getProductById({
        pathParameters: {
            productId: '5ecff64a-16c4-4f04-a68b-9dbd9f84dd7e',
        }
    } as unknown as APIGatewayProxyEvent, null, null);
    expect(result).toMatchSnapshot();
});

it('should reject with 404 error', async () => {
    const result = await getProductById({
        pathParameters: {
            productId: '83c8709f-23fb-465f-bc96-9e35f33554e9',
        }
    } as unknown as APIGatewayProxyEvent, null, null);
    expect(result).toMatchSnapshot();
});

