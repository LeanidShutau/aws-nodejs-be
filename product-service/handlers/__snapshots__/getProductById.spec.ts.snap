// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should reject with 404 error 1`] = `
Object {
  "body": "{
  \\"message\\": \\"Not Found\\"
}",
  "statusCode": 404,
}
`;

exports[`should response with a product 1`] = `
Object {
  "body": "{
  \\"product\\": {
    \\"id\\": \\"5ecff64a-16c4-4f04-a68b-9dbd9f84dd7e\\",
    \\"title\\": \\"Canned air from Paris\\",
    \\"description\\": \\"Canned air from Paris\\",
    \\"price\\": 14,
    \\"count\\": 10,
    \\"image\\": \\"https://i.etsystatic.com/6020270/r/il/48e8c7/2400885419/il_794xN.2400885419_jsos.jpg\\"
  }
}",
  "statusCode": 200,
}
`;
